Sustainable society hackathon materials

lisancao opened the pull request.

lisancao committed:
initial commit

lisancao committed:
refactor, add data, add code

lisancao committed:
add pets and tutorial files

lisancao committed:
file refactor, add notebook

lisancao committed:
add question 1A

lisancao committed:
add data in directory, complete challenge 1

lisancao committed:
moved challenge interactives to py script, minor grammar fixes and add fill in code

lisancao committed:
begin challenge 2, add files and functions

alyamahmoud committed:
hackathon templates

alyamahmoud committed:
Merge branch 'master' of https://github.com/callysto/hackathon

alyamahmoud committed:
hackathon templates

alyamahmoud committed:
hackathon example 01

lisancao committed:
add text colouring to Q1 and embed youtube videos

lisancao committed:
add Challenge2 code and youtube videos

lisancao committed:
more content, sectioned py script better

lisancao committed:
add tutorial, more content

- add place answer below line for cells
- add challenge 2b question
- added copy of pandas cheatsheet to directory

lisancao committed:
directory rename to camelcase

lisancao committed:
code fix and temp cell added

lisancao committed:
add outline for rest of notebook

lisancao committed:
cleared output

lisancao committed:
add notebook-basics

lisancao committed:
add challenge2c, content, minor fixes

lisancao committed:
small content additions

lisancao committed:
add content

alyamahmoud committed:
Rename hackathon_example_01_music.ipynb to music-example.ipynb

alyamahmoud committed:
CamelCase for directories and kebab-case for files

alyamahmoud committed:
Create 1

alyamahmoud committed:
Add files via upload

CamelCase for directories and kebab-case for files, directory structure

alyamahmoud committed:
Delete 1

alyamahmoud committed:
Create 1

alyamahmoud committed:
Add files via upload

CamelCase for directories and kebab-case for files, directory structure

alyamahmoud committed:
Delete 1

alyamahmoud committed:
update example notebook

alyamahmoud committed:
update template notebook

alyamahmoud committed:
Add files via upload

adding image to music example notebook

alyamahmoud committed:
added music image

alyamahmoud committed:
Merge remote-tracking branch 'origin/sustainable-society'

alyamahmoud committed:
update example notebook

alyamahmoud committed:
Add files via upload

add plastics example image

alyamahmoud committed:
added plastic example image

lisancao committed:
add content, refactor

- questions now output results
- remove answer cells
- clear output on long interactives
- add content

lisancao committed:
Merge branch 'sustainable-society' of https://github.com/callysto/hackathon into sustainable-society

lisancao committed:
add challenge 3

lisancao committed:
Add content

lisancao committed:
aesthetic rehaul

lisancao committed:
add text formatting functions

lisancao committed:
text encoding overhaul

lisancao committed:
add more content

lisancao committed:
add display() functionality, code clean, bugfix, add content

lisancao committed:
add basic stats content

lisancao committed:
complete challenge 4

lisancao committed:
structural changes, content fixes, add bonus challenge

lisancao committed:
add comments, bugfix q4c not executing

lisancao committed:
challenge4c hotfix

lisancao committed:
challenge4d hotfix

lisancao committed:
userinfo tweaks

alyamahmoud committed:
adding plastic dataset

alyamahmoud committed:
updating plastic notebook

alyamahmoud committed:
Merge remote-tracking branch 'origin/sustainable-society'

alyamahmoud committed:
updating hackathon templates and examples

alyamahmoud committed:
update plastics dataset

alyamahmoud committed:
update plastics dataset

alyamahmoud committed:
Delete plastic-example-image.jpeg

alyamahmoud committed:
Add files via upload

alyamahmoud committed:
Delete plastic-example-image.jpg

alyamahmoud committed:
Add files via upload

alyamahmoud committed:
Delete plastic-example-image.jpg

alyamahmoud committed:
Add files via upload

alyamahmoud committed:
Delete plastic-example-image.jpg

alyamahmoud committed:
Add files via upload

lisancao committed:
add content

lisancao committed:
add content

alyamahmoud committed:
update plastics exmaple notebook

lisancao committed:
Merge branch 'sustainable-society' of https://github.com/callysto/hackathon into sustainable-society

lisancao committed:
add content

lisancao committed:
Split notebooks, add images

- split notebooks based on challenges
- add images to images folder
- replace iframe with YouTubeVideo
- minor bugfix preventing challenge1A from starting
- commented out code for custom plots (recursion issue)
- add star emoji separator at end of challenge complete
- prompt for challenge complete cell to be edited with answers

lisancao committed:
file cleanup

alyamahmoud committed:
Delete plastic-example-image.jpg

alyamahmoud committed:
Add files via upload

alyamahmoud committed:
Delete plastic-example-image.jpg

alyamahmoud committed:
Add files via upload

alyamahmoud committed:
Delete plastic-example-image.jpg

alyamahmoud committed:
Add files via upload

alyamahmoud committed:
cufflinks for plots

alyamahmoud committed:
notebook update

alyamahmoud committed:
Delete plastic-example-image.jpg

alyamahmoud committed:
Add files via upload

alyamahmoud committed:
working on readability

alyamahmoud committed:
template update - readability level

alyamahmoud committed:
updating plastic example notebook

alyamahmoud committed:
updating plastic notebook

alyamahmoud committed:
updating music notebook

alyamahmoud committed:
cufflinks

alyamahmoud committed:
cufflinks

lisancao committed:
bugfix challenge2b

lisancao committed:
remove bitly link

byrcyb commented on SustainabilityOnMars/GuidedNotebooks/pets-in-space/notebook-basics.ipynb:
```diff
@@ -0,0 +1,124 @@
+    "Once you have logged in first time - your personal environment will be created, where all your work will be stored.\n",
+    "\n",
+    "\n",
+    "### Copy notebooks to Callysto Hub\n",
```
I would delete the "Copy notebooks to Callysto Hub" section as it's probably too advanced for how the students will interact with the notebooks. And the bitly link won't work properly.

byrcyb commented on SustainabilityOnMars/GuidedNotebooks/pets-in-space/challenge-0.ipynb:
```diff
@@ -0,0 +1,137 @@
+   "source": [
+    "![Callysto.ca Banner](https://github.com/callysto/curriculum-notebooks/blob/master/callysto-notebook-banner-top.jpg?raw=true)\n",
+    "\n",
+    "<a href=\"https://hub.callysto.ca/jupyter/hub/user-redirect/git-pull?repo=https%3A%2F%2Fgithub.com%2Fcallysto%2Fhackathon&branch=master&subPath=ColonizingMars/ChallengeTemplates/challenge-option-1-should-we-colonize-Mars.ipynb&depth=1\" target=\"_parent\"><img src=\"https://raw.githubusercontent.com/callysto/curriculum-notebooks/master/open-in-callysto-button.svg?sanitize=true\" width=\"123\ ...
```
This url is wrong and if the students click on it it will take them to another notebook. I imagine all the open in Callysto buttons need updating in this `SustainabilityOnMars` directory.

byrcyb replied:
we can update these once we merge them to master

lisancao committed:
hotfix for challenge3d

byrcyb commented on SustainabilityOnMars/GuidedNotebooks/pets-in-space/challenge-0.ipynb:
```diff
@@ -0,0 +1,137 @@
+    }
+   ],
+   "source": [
+    "YouTubeVideo('P6KNkb7xVjk') "
```
I think you did this on purpose so they would have the video without having to run the first cell but note I didn't see the video when I first opened my notebook in the CallystoHub. If you want you can just move this line onto the above code block.

byrcyb commented on SustainabilityOnMars/GuidedNotebooks/pets-in-space/challenge-0.ipynb:
```diff
@@ -0,0 +1,137 @@
+   "source": [
+    "## 📚 Challenge Complete!\n",
+    "\n",
+    "Wow, great job! Now that you've gotten started, continue to Challenge 1 in the next notebook!"
```
You can hyperlink to the next notebook so "... continue to [Challenge 1 in the next notebook](./challenge-1.ipynb)!"

lisancao committed:
challenge 5d hotfix

byrcyb commented on SustainabilityOnMars/GuidedNotebooks/pets-in-space/challenge-1.ipynb:
```diff
@@ -0,0 +1,148 @@
+    "### 📕 Debrief: Functions, Libraries, and Filenames\n",
+    "Now that we have access to the special commands we need to access the data, let's use them! The data file they gave us is called `pets_from_bootstrap_world.csv` - the .csv at the end tells us it is a csv file. The CSV stands for \"Comma Separated Values\", there is a special function in the `pandas` library we can use to view this data file called `read_csv()`. \n",
+    "\n",
+    "Python is a little tricky though, and needs us to put in the code which library the function we want to use comes from. For example, if I wanted to use the `loadtxt()` function from the `numpy` library, I would have to write it like `numpy.loadtxt('file.txt')`. See how the library is first, then there is a `.`, followed by the function? We then use the function on the file by putting the fil ...
```
Python is a little tricky though, and needs us to write the code to tell it which library the function we want to use comes from.

lisancao committed:
spelling errors

lisancao committed:
readability changes

byrcyb commented on SustainabilityOnMars/GuidedNotebooks/pets-in-space/challenge-1.ipynb:
```diff
@@ -0,0 +1,148 @@
+    "## 📚 Challenge Complete!\n",
+    "Wow, great job! You've completed the very first challenge with flying colours. Now, before moving onto the next challenge, write down the age of the pets in rows `28` (Miaulis), `15` (Gir), and `11` (Maple). **Make sure you write it somewhere that's safe, because you never know when it will come in handy! ;)**\n",
+    "\n",
+    "#### ✏️    DOUBLE CLICK HERE AND PROVIDE YOUR ANSWER    ✏️"
```
add hyperlink to next challenge "... [next challenge](./challenge-2.ipynb) ...."

byrcyb reviewed (commented):
Only made it to the end of Challenge 1 but looks good so far.

lisancao committed:
remove copy notebooks bits

lisancao committed:
add next notebook links, minor content edits

byrcyb commented:
@lisancao Some additional review comments:
- in Challenge 2b the answer is "columns" but it should be ".columns"
- in Challenge 3c to build off 3b and the debrief (just above challenge 3c) maybe the answer should be `pets.loc[pets["Legs"]==3]` or maybe change the debrief description so that there's an example like after "We can get the actual rows of data by using by using dataset.loc[label]" such as getting data from row 30 you would use `dataset.loc[30]
- in Challenge 3d, the solution has an extra set of round brackets that you don't introduce before so I would remove them. You don't necessarily need them as this will work too: solution [(pets["column_name"] == "filter_word")] vs [pets["column_name"] == "filter_word"]

byrcyb commented:
@lisancao Some last comments 😄 
- challenge 5, in the debrief: visualization markdown cell "y piece is the data that changes because of y" should be "y piece is the data that changes because of x"
- challenge 5d the solution that pops up when you enter a wrong answer says `plot.scatter(pets['Age (years)'], pets['Weight (lbs)'])` but it should plt not plot so `plt.scatter(pets['Age (years)'], pets['Weight (lbs)'])`

There's also a separate a separate notebook here (/GuidedNotebooks/pets-in-space/other/pets-in-space.ipynb) that I think is all the notebooks combined along with a bonus challenge that I didn't look at and can maybe be deleted?

Great job on the notebooks, I really like how each challenge builds off of each other and the feedback they get!

byrcyb commented:
@alyamahmoud for challenge-template.ipynb, I would make the following edits:
- Modify this sentence to "Use as many code cells as you need to analyze and visualize your data - remember to add a title, as well as appropriate x and y labels to your visualizations. 
- Change "Ensure to briefly comment on what you see." to "Make sure you write down what things you notice in the data such as trends, patterns, and basic statistics."
- Change "A sample is provided." to "Use the code cell below to start adding your code for data analysis and visualization"
- Change "Provide your analysis-driven observation(s) here." to "Double-click this cell and write down what things you observed in your data through your analysis and visualizations."
- Change "Provide your analysis-driven result(s) here. Tell us 2-3 things your leanrt from the data. Double-click to edit this cell. " to "Double-click this cell and write down 2 - 3 things you learned from the data and how they help answer your main question(s). "

byrcyb commented:
@alyamahmoud for your music-example.ipynb notebook, your youtube code cell won't work in the [Callysto Hub](hub.callysto.ca) without having everyone install the `youtube_search` package. To avoid this I would replace your code block with this
```
from IPython.display import YouTubeVideo
YouTubeVideo('JT7oaYFEgJw')
```

- I would also make sure you clear all the output from your cells before committing the notebook again to github (from the menu Cell -> All Output -> Clear)
- Overall it looks great!

byrcyb commented:
@alyamahmoud for your plastic-example.ipynb notebook
- Generally don't put a space before punctuation. For example with your question marks " ?" -> "?"
- Edit "... where does tha mjority of plastic waste come from ? we ..." -> "... where does the majority of plastic waste come from? We ..."
- Add a period to the end of the sentence that starts with "2. Ocean Conservancy TIDES Database:"
- Code cell that imports the data. You already have the data in the notebook directory so you don't need to read it in using a url unless you are doing this on purpose. If this is on purpose and you are showing them how to import from url then that's ok. Same thing for your music-example.ipynb notebook. Also I don't believe you need to specify `index_col=0` in either notebook as that's the default. If you are to switch to the local dataset you can use
`pd.read_csv("plastics-data-2015-2018.csv", low_memory=False)
`
- Edit typos "Let's breakdown this number and look at the contribution of each collected plastic item that led to Ghanad having the highest number of collected plastic itesm. We will draw a barplot to visualise the collected plastic items and their counts." -> "Let's breakdown this number and look at the contribution of each collected plastic item that led to Ghana having the highest number of collected plastic items. We will draw a barplot to visualise the collected plastic items and their counts."
- For when you find the highest number of plastic items, I would sequence it like this to avoid using argmax which they probably won't know about (and we don't reference)
```
# find largest number of plastic items collected and classified
plastics['TotalClassifiedItems_EC2020'].max()
```
then Markdown cell with:
The highest number of plastic items collected was 26420613. That's a lot! 
Next, let's find out more about where this plastic was collected from.
```
# First let's sort the data by the 'TotalClassifiedItems_EC2020' column by descending order i.e. highest is first
# Note the row number associated with the highest items which is 50185
plastics.sort_values(by="TotalClassifiedItems_EC2020", ascending = False)
```
```
# Now let's have a closer look at the data associated with the highest plastic items, row number 50185
plastics.loc[50185]
```

- Under the first "Observations" heading when you create a DataFrame, I would use this syntax as it something we typically share with students
```
maximum_plastics_waste = pd.DataFrame(plastics.loc[50185, 'SUM_Hard_PlasticBeverageBottle':'SUM_OtherPlasticDebris'])
```
- In this sentence debris is misspelled "We can see here that the biggest sources of plastic pollution in Ghana, comes from foam plastic debirs and foam. Cigarette butts, and soft straw also contribute plastic waste sources in Ghana."
- Grammar and typo corrections "### What about lollipop sticks? Do we have to give up lollipops on Mars?" 
- Typos corrections "Notice that lollipop sticks and ear buds are counted and classified among the plastic items that contribute to plastic waste. We would like to answer the question of how much do lollipop sticks actually contribute to plastic waste? "
- Typos in title of graph 'How much does lollipops actually contirubte to plastic waste ?' -> "'How much do lollipop sticks contribute to plastic waste?'
- Typos and grammar corrections "We observe that Europeans have lollipop sticks (and ear buds) as contributing to plastic waste much more than other continents.

Does this make sense? Does this mean people in all continents but Europe, don't eat lollipops? 

Of course NOT! People all over the globe eat lollipops."

- Overall, great dataset find and great narrative! A nice way to introduce them to how you need to question datasets before jumping to conclusions 😄

misterhay commented:
These notebooks are great, only a few minor changes required (as Byron has indicated). I'd also suggest changing the title in the first cell of `challenge-template.ipynb` to something like *Sustaining Life on Mars: Data Science Challenge* to be similar to the wording in the event invitation.

Perhaps we should also change the sentences that start with "Pick as many questions from the above section" to something like "Choose one or more of these questions to answer, or come up with your own. Check out the [example notebooks](https://github.com/callysto/hackathon/tree/sustainable-society/SustainabilityOnMars/ChallengeExamples), and complete the sections in this notebook to answer your chosen question or questions."

Under the *About You* section, I think we can just have the example list and delete the list that starts with "1. your name". As well, for this notebook do we want to change the pronouns from "your" to "my" since it will be the students writing it? For example, "About Me" and "The Data I Used".

These notebooks are looking good.

lisancao committed:
Changes based on review

- removed round brackets in challenge 3
- add sentence to debrief in challenge 3
- pre-run video cells
- spelling/grammar errors in challenge 5
- remove other folder

lisancao committed:
Changes based on review

- removed round brackets in challenge 3
- add sentence to debrief in challenge 3
- pre-run video cells
- spelling/grammar errors in challenge 5

lisancao committed:
Merge branch 'sustainable-society' of https://github.com/callysto/hackathon into sustainable-society

alyamahmoud committed:
wording fix

alyamahmoud committed:
Merge remote-tracking branch 'origin/sustainable-society'

alyamahmoud committed:
clear output

alyamahmoud committed:
wording fix

alyamahmoud committed:
YouTubeVideo

alyamahmoud committed:
punctuation fix

alyamahmoud committed:
rewording

alyamahmoud committed:
wording fix

alyamahmoud committed:
wording fix

alyamahmoud committed:
wording fix

alyamahmoud committed:
punctuation fix

alyamahmoud committed:
title fix

alyamahmoud committed:
title fix

alyamahmoud committed:
typos

alyamahmoud committed:
typos

alyamahmoud committed:
typos

byrcyb committed:
Update challenge-template.ipynb

Update instructions in observations and conclusions section of template.

byrcyb reviewed (approved):
All changes look good @alyamahmoud and @lisancao. FYI, I made a quick edit to the challenge template wording and contributed that as well.
